R/settings.R

```diff
@@ -164,7 +174,11 @@ validate_config_file <- function(config, config_file, defaults) {
   matched <- names(config) %in% names(defaults)
   if (!all(matched)) {
     unused_settings <- names(config)[!matched] # nolint: object_usage_linter. TODO(#2252).
-    cli_warn("Found unused settings in config {.str config_file}: {.field unused_settings}")
+    config_link <- cli::style_hyperlink( # nolint: object_usage_linter. TODO(#2252).
+      cli::col_blue(basename(config_file)),
+      url = paste0("file://", config_file)
+    )
```

Comment:
Should we create a helper to create these clickable links? 

We are already doing it twice in `settings.R`, and maybe will need to do it a few more times in future, and I don't wish for these `nolint` directives to proliferate.

Reply:
Moved it to helper. However, still need the nolint directive until #2252 is fixed. It is that the variable created in unused, except in the error message.
The nolint can only be removed if the function is called directly in the cli message